app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/metrics_capture.rb

```diff
@@ -0,0 +1,158 @@
+      :rollup                => "average",
+      :unit_key              => "kilobytespersecond",
+      :capture_interval_name => "realtime",
+    }.freeze,
+    "mem_usage_absolute_average" => {
+      :counter_key           => "mem_usage_absolute_average",
+      :instance              => "",
+      :capture_interval      => "20",
+      :precision             => 1,
+      :rollup                => "average",
+      :unit_key              => "percent",
+      :capture_interval_name => "realtime",
+    }.freeze,
+  }.freeze
+
+  def perf_collect_metrics(interval_name, start_time = nil, end_time = nil)
+    require 'rest-client'
+
+    raise _("No EMS defined") if ext_management_system.nil?
+
+    end_time ||= Time.zone.now
+    end_time   = end_time.utc
+    start_time ||= end_time - 4.hours
+    start_time   = start_time.utc
+
+    counters_by_mor = {target.ems_ref => VIM_STYLE_COUNTERS}
+    counter_values_by_mor = {target.ems_ref => {}}
+
+    metrics_endpoint = ext_management_system.endpoints.find_by(:role => "metrics")
+    raise _("Missing monitoring instance id") if metrics_endpoint.nil?
+
+    instance_id = metrics_endpoint.options["monitoring_instance_id"]
+
+    response = RestClient::Request.execute(
+      :method  => :post,
+      :url     => "https://#{ext_management_system.provider_region}.monitoring.cloud.ibm.com/api/data",
+      :headers => {
+        'Content-Type'  => 'application/json',
+        'Authorization' => "Bearer #{iam_access_token}",
+        'IBMInstanceID' => instance_id
+      },
+      :payload => JSON.generate(get_metrics_query(target.name, end_time.to_i - start_time.to_i))
+    )
+    data = JSON.parse(response.body)
+    dataset = consolidate_data(data["data"])
+
+    store_datapoints!(dataset[:timestamps], dataset[:cpu_usage_rate_average], "cpu_usage_rate_average", counter_values_by_mor[target.ems_ref])
+    store_datapoints!(dataset[:timestamps], dataset[:mem_usage_absolute_average], "mem_usage_absolute_average", counter_values_by_mor[target.ems_ref])
+    store_datapoints!(dataset[:timestamps], dataset[:net_usage_rate_average], "net_usage_rate_average", counter_values_by_mor[target.ems_ref])
+    store_datapoints!(dataset[:timestamps], dataset[:disk_usage_rate_average], "disk_usage_rate_average", counter_values_by_mor[target.ems_ref])
+
+    return counters_by_mor, counter_values_by_mor
+  rescue RestClient::ExceptionWithResponse => err
+    log_header = "[#{interval_name}] for: [#{target.class.name}], [#{target.id}], [#{target.name}]"
+    _log.error("#{log_header} Unhandled exception during perf data collection: [#{err}], class: [#{err.class}]")
+    _log.log_backtrace(err)
+    raise
+  end
+
+  def get_metrics_query(instance_name, time_window)
+    # :last refers to the window of time in which data will be retrieved
+    # :sampling refers to the data resolution, data will be sampled every 60 seconds
+    {
+      :last           => time_window,
+      :sampling       => 60,
```

Comment:
Since we're requesting every 60sec but our metrics backend is expecting 20sec (https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/308/files#diff-33defe66ca0940649be5af9d6676637d204a884b6016c5f39a736e2322c315c6R17) you'll have to interpolate each datapoint into three.  This is done in a few different providers already, also not sure if you could just request 20sec?

Reply:
E.g. see https://github.com/ManageIQ/manageiq-providers-azure/blob/master/app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb#L195-L198 and https://github.com/ManageIQ/manageiq-providers-google/blob/master/app/models/manageiq/providers/google/cloud_manager/metrics_capture.rb#L100-L112

Reply:
Sure, I'll look into this

Reply:
The API was not able to handle 20s sampling and so I've added the interpolation